integration-test.sh

```diff
@@ -470,13 +476,20 @@ function test_docker_dev_image {
 function test_node_management_commands {
     local cfg
     local dist
+    local build_type
     random_configuration cfg
     random_distribution dist
+    random_build_type build_type
+
+    # for Docker we force the most recent distribution as we don't have Docker images for all versions that are tested
+    if [[ "$build_type" == "docker" ]]; then
```

Comment:
While we kill Rally processes in `kill_related_es_processes` we don't do the same for any running Docker images. I think that we might need to stop Docker containers as well but I wonder whether we can find a robust approach of finding the correct image.

Reply:
This is possible. What we need to do is:

1. Change `docker-compose.yml.j2` to further specify a label like so:

```
diff --git a/esrally/resources/docker-compose.yml.j2 b/esrally/resources/docker-compose.yml.j2
index c5bc10a..19e72e9 100644
--- a/esrally/resources/docker-compose.yml.j2
+++ b/esrally/resources/docker-compose.yml.j2
@@ -4,6 +4,8 @@ services:
     cap_add:
       - IPC_LOCK
     image: "{{docker_image}}:{{es_version}}"
+    labels:
+      io.rally.description: "Label to help identify Elasticsearch containers launched by Rally"
     {%- if docker_cpu_count is defined %}
     cpu_count: {{docker_cpu_count}}
     {%- endif %}
```

2. Use the label to retrieve the ID of container(s) launched by Rally using a filter like `docker ps --filter "label=io.rally.description" --format "{{.ID}}"` (if >1 they'll be newline separated).
    This will match all nodes matching the specific label key. To more targeted, we can also match the value which could be optionally specified via a j2 variable in the compose file, but this functionality would require an optional argument for the start subcommand (e.g. `--docker-label="..."`) to override the j2 variable.

Reply:
Thanks! I'll implement this in the integration test. I think it's fine to use the less targeted approach and rely on a static label in `docker-compose.yml.j2`.

Reply:
I agree.

Reply:
Added in https://github.com/elastic/rally/pull/830/commits/9ef5660278c22ebdb88587e410dabf13b98711d9.